core/src/tasks/update_wallet.rs

```diff
@@ -323,172 +286,51 @@ impl UpdateWalletTask {
+    async fn add_new_orders_to_book(&self) {
+        let local_cluster_id = self.global_state.local_cluster_id.clone();
+        let wallet_public_share_nullifier = self.new_wallet.get_public_share_nullifier();
 
-    /// Update the order in the state
-    async fn update_order_state(
-        &self,
-        order_id: OrderIdentifier,
-        proof: ValidCommitmentsBundle,
-        witness: SizedValidCommitmentsWitness,
-    ) -> Result<(), UpdateWalletTaskError> {
-        // If the order does not currently exist in the book, add it
-        if !self
-            .global_state
-            .read_order_book()
-            .await
-            .contains_order(&order_id)
-        {
-            self.global_state
-                .add_order(NetworkOrder {
-                    id: order_id,
-                    match_nullifier: self.new_wallet.get_match_nullifier(),
-                    local: true,
-                    cluster: self.global_state.local_cluster_id.clone(),
-                    state: NetworkOrderState::Verified,
-                    valid_commit_proof: Some(proof.clone()),
-                    valid_commit_witness: Some(witness),
-                })
-                .await;
-        } else {
-            // Otherwise, update the existing proof
-            self.global_state
-                .add_order_validity_proof(&order_id, proof.clone())
-                .await;
-            self.global_state
+        for order_id in self.new_wallet.orders.keys() {
+            if !self
+                .global_state
                 .read_order_book()
                 .await
-                .attach_validity_proof_witness(&order_id, witness)
-                .await;
+                .contains_order(order_id)
+            {
+                self.global_state
+                    .add_order(NetworkOrder::new(
+                        *order_id,
+                        wallet_public_share_nullifier,
+                        local_cluster_id.clone(),
+                        true, /* local */
+                    ))
+                    .await
+            }
         }
+    }
 
-        // Broadcast the new order's validity proof to the network
-        let cluster = self.global_state.local_cluster_id.clone();
-        let message = OrderBookManagementMessage::OrderProofUpdated {
-            order_id,
-            cluster,
-            proof,
-        };
-
-        self.network_sender
-            .send(GossipOutbound::Pubsub {
-                topic: ORDER_BOOK_TOPIC.to_string(),
-                message: PubsubMessage::OrderBookManagement(message),
-            })
-            .map_err(|err| UpdateWalletTaskError::SendMessage(err.to_string()))
+    /// After a wallet update has been submitted on-chain, find its authentication
+    /// path, and re-prove `VALID REBLIND` for the wallet and `VALID COMMITMENTS`
+    /// for all orders in the wallet
+    async fn update_validity_proofs(&self) -> Result<(), UpdateWalletTaskError> {
```

Comment:
maybe i'm blind, but do we actually call this after submitting the updated wallet on-chain?

Reply:
ah, a function w this name existed before, i take it we do call this it just doesn't show up in the diff

Reply:
that's correct, call happens in state transition [here](https://github.com/renegade-fi/renegade/blob/main/core/src/tasks/update_wallet.rs#L172)